src/VectorizeLoops.cpp

```diff
@@ -727,26 +717,30 @@ class VectorSubs : public IRMutator {
                     Stmt without_likelies =
                         IfThenElse::make(op->condition.as<Call>()->args[0],
                                          op->then_case, op->else_case);
-                    stmt =
+                    Stmt stmt =
                         IfThenElse::make(all_true,
                                          then_case,
                                          scalarize(without_likelies));
                     debug(4) << "...With all_true likely: \n" << stmt << "\n";
+                    return stmt;
                 } else {
-                    stmt =
+                    Stmt stmt =
                         IfThenElse::make(all_true,
                                          then_case,
                                          predicated_stmt);
                     debug(4) << "...Predicated IfThenElse: \n" << stmt << "\n";
+                    return stmt;
                 }
             } else {
                 // It's some arbitrary vector condition.
                 if (!vectorize_predicate) {
-                    debug(4) << "...Scalarizing vector predicate: \n" << stmt << "\n";
-                    stmt = scalarize(op);
+                    Stmt stmt = scalarize(op);
```

Comment:
This changes the order in which debugging messages are printed.

Reply:
True, but 'stmt' was undefined at this point in the old code, so this printed nothing.

Reply:
I think that issue is independent -- I was thinking that if there are debugging prints done inside scalarize(...), then in the old code these gets printed after "scalarizing vector predicate...", and in the new code it gets printed before that. But the stmt arg should indeed have been something like Stmt(op). I think printing op directly might print it as a pointer.

Reply:
You are correct. Fixed.